### ISSUE TITLE
Cache-control header is filled incorrectly

### ISSUE THREAD

**Issue by ikoveshnikov** [buildbot/buildbot#5220]
When a cache-control header is formed a ';' character is used as a delimiter:

https://github.com/buildbot/buildbot/blob/144eb7e82dc261e6506f1f68493446bcb24d77a0/master/buildbot/www/config.py#L120

This is not allowed by [RFC 7234](https://tools.ietf.org/html/rfc7234). The RFC states the following format of the header:
```
Cache-Control = *( "," OWS ) cache-directive *( OWS "," [ OWS
    cache-directive ] )
 cache-directive = token [ "=" ( token / quoted-string ) ]
```

Thus a replace `;` -> `, ` is required.

**Comment by pmisik** [comment]
Hi @ikoveshnikov 
To me it seems you are right. 

I just wonder what issue you have observed you figured this problem?
Any http proxy/browser issue you have seen?
It might be worth to mention in release notes if someone else has the same environment.

**Comment by ikoveshnikov** [comment]
Well, it happens on our web proxy server which is still under development, so I don't think anyone has the same environment. 

After I put it in front of buildbot I observed that buildbot responses are filtered out because of wrong `Cache-control` format. A little investigation showed me that our server follows RFC definitions and the buildbot response was malformed. In the same same time browsers and other proxy servers doesn't have such strict rules for `Cache-control` headers and buildbot website works well with them (tried: firefox, chrome, nginx). Probably they just ignore parsing errors in `Cache-control` header and treat response as non-cacheable, not really sure here.

So I came to conclusion that our server should be updated to follow not only the RFC but also behaviour of other servers/clients. And there is an issue in open-source project I cant report.
